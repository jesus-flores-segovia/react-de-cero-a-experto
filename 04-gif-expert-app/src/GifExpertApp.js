import { useState } from "react";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(["One Punch", "Samurai X", "Dragon Ball"]);

    const handleAdd = () => {
        setCategories([...categories, "Sword Art Online"]);
    };

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr/>

            <button onClick={handleAdd}>Add</button>

            <ol>
                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>;
                    })
                }
            </ol>
        </>
    );
};