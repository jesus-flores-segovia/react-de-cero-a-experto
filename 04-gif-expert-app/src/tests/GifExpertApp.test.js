import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import {GifExpertApp} from "../GifExpertApp";

describe("Tests inside file GifExpertApp.test.js", () => {

    test("'GifExpertApp' component must show correctly", () => {
        const wrapper = shallow(<GifExpertApp></GifExpertApp>)
        expect(wrapper).toMatchSnapshot();
    });

    test("'GifExpertApp' component must show a list of categories", () => {
        const categories = ["Dragon Ball", "One Punch"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}></GifExpertApp>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });
});