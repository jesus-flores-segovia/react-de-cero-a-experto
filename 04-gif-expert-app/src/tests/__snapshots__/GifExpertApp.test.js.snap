// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests inside file GifExpertApp.test.js 'GifExpertApp' component must show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
</Fragment>
`;

exports[`Tests inside file GifExpertApp.test.js 'GifExpertApp' component must show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
